### PR TITLE
Fix: Cancel heartbeat after browser window reload

### PR DESCRIPTION
**Pull request by xiekun1992**
[LU-2267](https://snapmaker2.atlassian.net/browse/LU-2267)
[LU-2268](https://snapmaker2.atlassian.net/browse/LU-2268)

### CODE REVIEW

**Review by jane-rose — APPROVED** [review]

File: src/app/flux/machine/Server.js [hunk 1]
```diff
@@ -346,4 +346,15 @@ export class Server extends events.EventEmitter {
         };
         return this.gcodePrintingInfo;
     }
+
+    static closeServerAfterWindowReload() {
```
Comment (jane-rose) [inline src/app/flux/machine/Server.js:350]: just using `this.closeServer`

File: src/server/services/socket/socket-http.ts [hunk 2]
```diff
@@ -403,10 +407,29 @@ class SocketHttp {
...
-                    msg: err?.message,
-                    text: res.text
-                });
+                if (err) {
+                    this.socket && this.socket.emit(eventName, {
+                        msg: err?.message,
+                        text: res.text
+                    });
+                } else {
+                    let gcodeStr = '';
```
Comment (jane-rose) [inline src/server/services/socket/socket-http.ts:416]: add this for sacp?

**Review by MaShizhen — APPROVED** [review]

File: src/app/flux/machine/Server.js (hunk 1 above)
Comment (xiekun1992) [reply src/app/flux/machine/Server.js:350]: static method can not access this, unless we create a Server instance just for doing this

File: src/server/services/socket/socket-http.ts (hunk 2 above)
Comment (xiekun1992) [reply src/server/services/socket/socket-http.ts:416]: no, just for HTTP

**Pull request merged**